File: Chapitre2/exercice2.1.c
#include <stdio.h>
#include <stdlib.h>
#include <errno.h>
#include <unistd.h>

int main()
{
    pid_t fils;
    fils=fork();
    if( fils == -1)
    {
        perror("Erreur lors de la création de processus");
        exit(1);
    }
    else if(fils == 0)
    {
        printf("Je suis dans le fils: moi=%d,père=%d\n",getpid(),getppid());
    }
    else
    {
        printf("Je suis le père: fils:%d,moi:%d,mon_père:%d\n", fils,getpid(), getppid());
    }

    return 0;
}

File: Chapitre2/exercice2.1.py
import os
import sys

pid = os.fork()

if pid == -1:
    print("Erreur lors de la création de processus", file=sys.stderr)
    sys.exit(1)
elif pid == 0:
    print(f"Je suis dans le fils: moi={os.getpid()},père={os.getppid()}")
else:
    print(f"Je suis le père: fils:{pid},moi:{os.getpid()},mon_père:{os.getppid()}")